### PR TITLE
Add support for building inside podman containers

### PR DESCRIPTION
**Pull request by abitrolly**
https://github.com/containers/libpod/blob/master/docs/podman-run.1.md

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/internal/common.py [hunk 1]
```diff
@@ -124,8 +126,8 @@ def is_snap() -> bool:
     return is_snap
 
 
-def is_docker_instance() -> bool:
-    return os.path.exists(_DOCKERENV_FILE)
+def is_in_container() -> bool:
```
Comment (sergiusens) [inline snapcraft/internal/common.py]: can you call this differently, the word container is a bit vague and certainly is not relevant to LXD containers

File: snapcraft/internal/common.py [hunk 2]
```diff
@@ -124,8 +126,8 @@ def is_snap() -> bool:
...
+    return os.path.exists(_DOCKERENV_FILE) or os.path.exists(_PODMAN_FLAG)
```
Comment (sergiusens) [inline snapcraft/internal/common.py]: How about
```suggestion
    return any([os.path.exists(p) for p in (_DOCKERENV_FILE, _PODMAN_FLAG)])
```

File: snapcraft/internal/common.py [hunk 3]
```diff
@@ -42,7 +42,9 @@
 _keyringsdir = _DEFAULT_KEYRINGSDIR
 _DEFAULT_LEGACY_SNAPCRAFT_DIR = os.path.join(sys.prefix, "legacy_snapcraft")
 _legacy_snapcraft_dir = _DEFAULT_LEGACY_SNAPCRAFT_DIR
+
 _DOCKERENV_FILE = "/.dockerenv"
+_PODMAN_FLAG = "/run/.containerenv"
```
Comment (sergiusens) [inline snapcraft/internal/common.py]: Why did you call this FLAG and not FILE?

File: snapcraft/internal/common.py (hunk 1 above)
Comment (abitrolly) [reply snapcraft/internal/common.py]: Ephemeral LXD containers behave the same. `def container_mode() -> bool` is ok? With explanation of what container mode is.

File: snapcraft/internal/common.py (hunk 3 above)
Comment (abitrolly) [reply snapcraft/internal/common.py]: Just to remind that it is just an empty file. Fixed.

Comment (abitrolly) [reply snapcraft/internal/common.py]: Renamed to `is_process_container` to be consistent with `is_snap` above and to distinguish between LXD full OS containers.

File: snapcraft/internal/common.py (hunk 1 above)
Comment (abitrolly) [reply snapcraft/internal/common.py]: Renamed to `is_process_container` to be consistent with `is_snap` above and to distinguish between LXD full OS containers.

**Comment by abitrolly** [comment]
@sergiusens rebased and ready to review.

**Pull request merged**